masq/src/commands/set_exit_location_command.rs

```diff
@@ -0,0 +1,224 @@
+
+const FALLBACK_ROUTING_HELP: &str = "If you just want to make a suggestion, and you don't mind Nodes in other countries being used if nothing is available \
+     in your preferred countries, you can specify --fallback-routing, and you'll get no error unless there are no exit Nodes \
+     available anywhere. \n Here are some examples: \n\n\
+     masq> exit-location --country-codes \"CZ\" --fallback-routing              // Set exit-location for \"CZ\" country with fallback-routing on \n\
+     masq> exit-location --country-codes \"CZ\"                                 // Set exit-location for \"CZ\" country with fallback-routing off \n";
+
+pub fn set_exit_location_subcommand() -> App<'static, 'static> {
+    SubCommand::with_name("exit-location")
+        .about(EXIT_LOCATION_ABOUT)
+        .arg(
+            Arg::with_name("country-codes")
+                .long("country-codes")
+                .value_name("COUNTRY-CODES")
+                .validator(common_validators::validate_exit_locations)
+                .help(COUNTRY_CODES_HELP)
+                .required(false),
+        )
+        .arg(
+            Arg::with_name("fallback-routing")
+                .long("fallback-routing")
+                .value_name("FALLBACK-ROUTING")
+                .help(FALLBACK_ROUTING_HELP)
+                .takes_value(false)
+                .required(false),
+        )
+}
+
+pub mod tests {
+    use crate::commands::commands_common::{Command, STANDARD_COMMAND_TIMEOUT_MILLIS};
+    use crate::commands::set_exit_location_command::SetExitLocationCommand;
+    use crate::test_utils::mocks::CommandContextMock;
+    use masq_lib::messages::{
+        CountryCodes, ToMessageBody, UiSetExitLocationRequest, UiSetExitLocationResponse,
+    };
+    use std::sync::{Arc, Mutex};
+
+    #[test]
+    fn can_deserialize_ui_set_exit_location() {
+        let expected_request = UiSetExitLocationRequest {
+            fallback_routing: true,
+            exit_locations: vec![
+                CountryCodes {
+                    country_codes: vec!["CZ".to_string(), "SK".to_string()],
+                    priority: 1,
+                },
+                CountryCodes {
+                    country_codes: vec!["AT".to_string(), "DE".to_string()],
+                    priority: 2,
+                },
+                CountryCodes {
+                    country_codes: vec!["PL".to_string()],
+                    priority: 3,
+                },
+            ],
+        };
+        let transact_params_arc = Arc::new(Mutex::new(vec![]));
+        let mut context = CommandContextMock::new()
+            .transact_params(&transact_params_arc)
+            .transact_result(Ok(UiSetExitLocationResponse {}.tmb(0)));
+        let stderr_arc = context.stderr_arc();
+        let subject = SetExitLocationCommand::new(&[
+            "exit-location".to_string(),
+            "--country-codes".to_string(),
+            "CZ,SK|AT,DE|PL".to_string(),
```

Comment:
Q. What if there are spaces in this string?
A. Then it becomes several parameters to `--country-codes`.

Whoa!

If we could provide several parameters to `--country-codes`, we might be able to have more straightforward syntax, and not have to rely on commas and pipes. What about `--country-codes CZ,SK AT,DE PL`? Is that clearer?